client/express-checkout/event-handler.js

```diff
@@ -122,6 +124,89 @@ export const onConfirmHandler = async (
 	}
 };
 
+export const onConfirmHandlerForBlocksAPI = async (
+	api,
+	stripe,
+	elements,
+	completePayment,
+	abortPayment,
+	event,
+	order = 0 // Order ID for the pay for order flow.
+) => {
+	const submitResponse = await elements.submit();
+	if ( submitResponse?.error ) {
+		return abortPayment( event, submitResponse?.error?.message );
+	}
+
+	const { paymentMethod, error } = await stripe.createPaymentMethod( {
+		elements,
+	} );
+
+	if ( error ) {
+		return abortPayment( event, error.message );
+	}
+
+	try {
+		// Kick off checkout processing step.
+		let orderResponse;
+		if ( ! order ) {
+			orderResponse = await api.expressCheckoutECECreateOrderForBlocksAPI(
+				normalizeOrderDataForBlocksAPI( event, paymentMethod.id )
+			);
+		} else {
+			orderResponse = await api.expressCheckoutECEPayForOrderForBlocksAPI(
+				order,
+				normalizePayForOrderDataForBlocksAPI( event, paymentMethod.id )
+			);
+		}
+
+		if ( orderResponse.payment_result.payment_status !== 'success' ) {
+			return abortPayment(
+				event,
+				getErrorMessageFromNotice(
+					orderResponse.payment_result?.payment_details.find(
+						( detail ) => detail.key === 'errorMessage'
+					)?.value
+				),
+				true
+			);
+		}
+
+		const confirmationRequest = api.confirmIntent(
+			orderResponse.payment_result.redirect_url
+		);
+
+		// `true` means there is no intent to confirm.
+		if ( confirmationRequest === true ) {
+			completePayment( orderResponse.payment_result.redirect_url );
+		} else {
+			const { request } = confirmationRequest;
+			const redirectUrl = await request;
+
+			completePayment( redirectUrl );
+		}
+	} catch ( e ) {
+		let errorMessage;
+		if ( e.message ) {
+			errorMessage = e.message;
+		} else {
+			const paymentDetailsErrorMessage = e.payment_result?.payment_details.find(
+				( detail ) => detail.key === 'errorMessage'
+			)?.value;
+			if ( paymentDetailsErrorMessage ) {
+				errorMessage = paymentDetailsErrorMessage;
+			}
+		}
+		if ( ! errorMessage ) {
+			errorMessage = __(
+				'There was a problem processing the order.',
+				'woocommerce-gateway-stripe'
+			);
+		}
+		return abortPayment( event, getErrorMessageFromNotice( errorMessage ) );
```

Comment:
```suggestion
		return abortPayment( event, getErrorMessageFromNotice( errorMessage ) , true);
```

I came across a scenario where I was trying to pay with GPay when I was logged out as a customer. I got this error notice which is expected.

<img width="963" alt="Screenshot 2025-01-10 at 12 31 18 AM" src="https://github.com/user-attachments/assets/9087198e-aace-45d7-8131-0ec2ce769083" />

But also got the following error in the console which goes away if you set the third param as `true`.

![Uploading Screenshot 2025-01-10 at 12.25.53 AM.png…]()

Reply:
Good catch! Suggestion accepted in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3691/commits/003bce9dc1f64fcdecc9e7d44edb0dc7386213b5